Node2Code_Engine/Query/Statements.cs

```diff
@@ -0,0 +1,327 @@
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BH.Engine.Node2Code
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        public static List<StatementInfo> IStatements(this INode node, Dictionary<Guid, Variable> variables, int depth = 0)
+        {
+            if (node.IsInline)
+            {
+                List<ExpressionSyntax> arguments = Arguments(node, variables, out List<ReceiverParam> listInputs);
+                if (listInputs.Count > 0 && !(node is SetPropertyNode)) // need a better way to do this
+                    node.Outputs.ForEach(output => PromoteToList(variables, output.BHoM_Guid));
+                return new List<StatementInfo>();
+            }
+            else if (node is BlockNode)
+                return Statements(node as BlockNode, variables, depth);
+            else
+                return Statements(node, variables);
+        }
+
+
+        /***************************************************/
+        /**** Private Methods                           ****/
+        /***************************************************/
+
+        public static List<StatementInfo> Statements(this BlockNode node, Dictionary<Guid, Variable> variables, int depth = 0)
+        {
+            string commentLead = new string(Enumerable.Repeat('-', 4 * depth).ToArray());
+
+            List<StatementInfo> statements = node.InternalNodes.SelectMany(x => x.IStatements(variables, depth + 1)).Where(x => x != null).ToList();
+
+            if (statements.Count > 0)
+            {
+                StatementSyntax statement = statements[0].Statement;
+                if (statement.HasLeadingTrivia)
+                {
+                    SyntaxTriviaList trivia = statement.GetLeadingTrivia();
+                    trivia = trivia.InsertRange(0, new List<SyntaxTrivia> {
+                        SyntaxFactory.Comment(" "),
+                        SyntaxFactory.Comment("//" + commentLead + " " + node.Description)
+                    });
+                    statements[0].Statement = statement.WithLeadingTrivia(trivia);
+                }
+                else
+                {
+                    statements[0].Statement = statements[0].Statement.WithLeadingTrivia(
+                        SyntaxFactory.Comment(" "),
+                        SyntaxFactory.Comment("//" + commentLead + " " + node.Description)
+                    );
+                }
+            }
+
+            return statements;
+        }
+
+        /***************************************************/
+
+        public static List<StatementInfo> Statements(this INode node, Dictionary<Guid, Variable> variables)
```

Comment:
Is this meant to be public as it's under the private comment above? If it is, then, Documentation?